pkg/crypto/protocol/difficulty.go

```diff
@@ -1,20 +1,25 @@
 package protocol
 
 import (
-	"encoding/binary"
-	"math/bits"
+	"crypto/sha256"
+	"encoding/hex"
+	"math/big"
 )
 
-// CountHashDifficultyBits returns the number of leading zero bits in the given byte slice.
-// TODO_MAINNET: Consider generalizing difficulty to a target hash. See:
+// Difficulty1Hash represents the "easiest" difficulty.
+var (
+	Difficulty1HashHex   = "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
```

Comment:
Adding godoc comments here. "Difficulty 1" is the "highest" possible hash, which corresponds to the lowest possible difficulty. I'm reluctant to use terms like `min` or `max` to avoid confusion as a result of this bizarre juxtaposition of highest and lowest. 

This is a value that we choose which is independent of the `DefaultRelayTargetHash`. It effectively calibrates the difficulty number (the return of `GetDifficultyFromHash`) by defining the hash which corresponds to the lowest difficulty: 1.